samples/graalvm-native-static-jna/build.gradle.kts

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2015 Adam Marcionek, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+plugins {
+  java
+  application
+  alias(libs.plugins.graalvm)
+}
+
+application {
+  mainClass = "com.example.JnaNative"
+}
+
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(22)
+        vendor = JvmVendorSpec.GRAAL_VM
+    }
+}
+
+dependencies {
+  implementation(libs.bundles.jna)
+  implementation(libs.bundles.graalvm.api)
+  nativeImageClasspath(libs.jna.graalvm)
+}
+
+val nativeImageDebug: String by properties
+
+graalvmNative {
+  testSupport = true
+  toolchainDetection = false
+
+  binaries {
+    named("main") {
+      buildArgs.addAll(listOf(
+        "--features=com.sun.jna.SubstrateStaticJNA",
+        "-J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.hosted.jni=ALL-UNNAMED",
+        "-J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.jni=ALL-UNNAMED",
+        "-J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.hosted=ALL-UNNAMED",
+        "-J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.hosted.c=ALL-UNNAMED",
+        "-J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk=ALL-UNNAMED",
```

Comment:
(these `jargs`)

Reply:
Would adding these by default not be a problem for downstream consumers who don't want to export such modules? I'm not sure whether that's a valid concern or not, on the other hand it would definitely be annoying if users needed to add this manually though.